test/Helper.cs

```diff
@@ -396,6 +397,161 @@ public static Transfer PatchTransferByIdTest()
 
+        public static Plan TestCreatePlan()
+        {
+            Dictionary<string, object> data = new Dictionary<string, object>();
+
+            using (StreamReader r = new StreamReader("./test/data/plan-create.json"))
+            {
+                string json = r.ReadToEnd();
+                data = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
+            }
+
+            return Helper.client.Plan.Create(data);
+        }
+
+        public static Plan TestFetchPlan()
+        {
+            Plan plan = TestCreatePlan();
+
+            return Helper.client.Plan.Fetch((string)plan["id"]);
+        }
+
+        public static List<Plan> TestFetchAllPlans()
+        {
+            return Helper.client.Plan.All();
+        }
+
+        public static Subscription TestCreateSubscription()
+        {
+            Dictionary<string, object> data = new Dictionary<string, object>();
+
+            using (StreamReader r = new StreamReader("./test/data/subscription-create.json"))
+            {
+                string json = r.ReadToEnd();
+                data = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
+            }
+
+            Subscription subscription = Helper.client.Subscription.Create(data);
+
+            return subscription;
+        }
+
+        public static Subscription TestFetchSubscription()
+        {
+            Subscription subscription1 = TestCreateSubscription();
+
+            Subscription subscription2 = Helper.client.Subscription.Fetch((string)subscription1["id"]);
+
+            Assert.AreEqual(subscription1["id"], subscription2["id"]);
+
+            return subscription2;
+        }
+
+        public static List<Subscription> TestFetchSubscriptions()
+        {
+            return Helper.client.Subscription.All();
+        }
+
+        public static Subscription TestCancelSubscription()
+        {
+            Subscription subscription = TestCreateSubscription();
+
+            return subscription.Cancel();
+        }
+
+        public static Addon TestCreateAddon()
+        {
+            Dictionary<string, object> data = new Dictionary<string, object>();
+
+            using (StreamReader r = new StreamReader("./test/data/addon-create.json"))
+            {
+                string json = r.ReadToEnd();
+                data = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
+            }
+
+            Subscription subscription = TestCreateSubscription();
+
+            return subscription.createAddon(data);
+        }
+
+        public static Addon TestFetchAddon()
+        {
+            Addon addon = TestCreateAddon();
+
+            Addon addon2 = Helper.client.Addon.Fetch((string)addon["id"]);
+
+            Assert.AreEqual(addon["id"], addon2["id"]);
+
+            return addon2;
+        }
+
+        // TODO: Need to re-test this
+        public static Addon TestDeleteAddon()
+        {
+            Addon addon = Helper.client.Addon.All()[0];

```

Comment:
we should be able to delete from a addon id. need to add a constructor to create addon with given Id